UnityProject/Assets/Scripts/Tilemaps/Behaviours/Meta/Atmospherics/Data/Gas.cs

```diff
@@ -68,33 +73,97 @@ public static Gas[] All
 			{
 				if (all == null)
 				{
-					all = gases.ToArray();
+					all = gases.Values.ToArray();
 				}
+
 				return all;
 			}
 		}
 
-
 		private static Gas[] all;
 
-
 		public static int Count
 		{
 			get
 			{
-				if (numberofGases == 0)
+				if (numberOfGases == 0)
 				{
-					numberofGases = gases.Count;
+					numberOfGases = gases.Count;
 				}
-				return numberofGases;
+				return numberOfGases;
 			}
 		}
 
-		private static int numberofGases = 0;
+		private static int numberOfGases = 0;
 
 		public static implicit operator int(Gas gas)
 		{
 			return gas.Index;
 		}
 	}
+
+	[Serializable]
+	public class GasData
+	{
+		//Used for quick iteration
+		public GasValues[] GasesArray = new GasValues[0];
+
+		//Used for fast look up for specific gases
+		public Dictionary<GasType, GasValues> GasesDict = new Dictionary<GasType, GasValues>( new GasTypeEnum());
+
+		public void RegenerateDict()
+		{
+			GasesDict.Clear();
+
+			for (int i = 0; i < GasesArray.Length; i++)
+			{
+				var value = GasesArray[i];
+				GasesDict.Add(value.GasType, value);
+			}
+		}
+	}
+
+	[Serializable]
+	public class GasValues
+	{
+		public GasType GasType;
+
+		//Moles of this gas type
+		public float Moles;
+	}
+
+	public enum GasType
```

Comment:
doesn't this mean you can't add new gases in runtime

Reply:
Theres two ways to make it work, either add some spare enum values at the end or just internally use int instead of the actual enum in the dictionary and then just keep a count on how many new added gases + enum total count to get a new id. Either way wouldnt be hard to allow for runtime gas creation in another PR

Reply:
Will probably remove the enum and just making the int auto generated, sounds easier.

Reply:
Actually will still need the enum for gas canister setting values but that'll just be to link to the gas type not for storing the data